src/html/product/review.html

```diff
@@ -73,5 +73,48 @@ <h1 class="page-title">商品のクチコミ</h1>
         </ul>
     </section>
 </main>
+
+<script>
+
+    const reviewTotal = document.getElementById("reviewTotal")
+
+    // li.starにClassを追加
+    function addStarClass(list,className,num){
+        for( let i in list){
+            if( list.hasOwnProperty(i) ) {
+                if( i == num ) break;
+                list[i].classList.add(className);
+            }
+        }
+    }
+    
+    function getStarScore(element){
+        const score = Math.round(element.dataset.score);
+        const stars = element.children;
+        console.log(score);
+        if( score == 0){
+            // score = 0
+            addStarClass(stars,'fill',0)
+        }else if( score >= 1 && score <= 20){
+            // score = 1-20
+            addStarClass(stars,'fill',1)
+        }else if( score >= 21 && score <= 40){
+            // score = 21-40
+            addStarClass(stars,'fill',2)
+        }else if( score >= 41 && score <= 60){
+            // score = 41-60
+            addStarClass(stars,'fill',3)
+        }else if( score >= 61 && score <= 80){
+            // score = 61-80
+            addStarClass(stars,'fill',4)
+        }else if( score >= 81 && score <= 100){
+            // score = 81-100
+            addStarClass(stars,'fill',5)
+        }else{
+            addStarClass(stars,'inactive',5)
+        }
+    }
     
+    getStarScore(reviewTotal);
```

Comment:
まるっとこんな感じでシンプルに置き換えられるので、一度みてみてください。
ES6の記述は古いAndroidとかで動かない恐れがあるので、今回は避けた方が良いと思う。

```suggestion
    var stars = document.querySelectorAll(".review-stars--total .star");
    var score = document.querySelector("#reviewTotal").dataset.score

    for(var i = 0; i < stars.length; i++) {   
        if (score / 100 * 5 < i + 1) {
            stars[i].classList.add("inactive")
        } else {
            stars[i].classList.add("fill")
        }
    }
```